packages/legacy-compat/src/legacy-network-handler/legacy-network-handler.ts

```diff
@@ -256,31 +256,40 @@ function adapterDidInvalidate(
   }
 }
 
-function makeArray<T>(value: T | T[]): T[] {
-  return Array.isArray(value) ? value : [value];
-}
-
 const PRIMARY_ATTRIBUTE_KEY = 'base';
-function errorsHashToArray(errors: Record<string, string | string[]>): JsonApiError[] {
+function errorsHashToArray(errorsHash: Record<string, string | string[] | JsonApiError[]>): JsonApiError[] {
   const out: JsonApiError[] = [];
 
-  if (errors) {
-    Object.keys(errors).forEach((key) => {
-      let messages = makeArray(errors[key]);
-      for (let i = 0; i < messages.length; i++) {
+  if (errorsHash) {
+    Object.keys(errorsHash).forEach((key) => {
+      let errorsItem = errorsHash[key]
+      let errors = typeof errorsItem === 'string' ? [errorsItem] : errorsItem;
+      for (let i = 0; i < errors.length; i++) {
         let title = 'Invalid Attribute';
         let pointer = `/data/attributes/${key}`;
         if (key === PRIMARY_ATTRIBUTE_KEY) {
           title = 'Invalid Document';
           pointer = `/data`;
         }
-        out.push({
-          title: title,
-          detail: messages[i],
-          source: {
-            pointer: pointer,
-          },
-        });
+        let error = errors[i]
+        if (typeof error === 'string') {
+          out.push({
+            title: title,
+            detail: error,
+            source: {
+              pointer: pointer,
+            },
+          });
+        } else {
+          out.push({
+            ...error,
+            title: title,
+            detail: error.detail || error.title,
+            source: {
+              pointer: pointer,
+            },
```

Comment:
If the full JsonApiError array was handed to this function, we produce an error that contains all the other (spec-complaint) error members, but we make sure it still has the title/detail/source as this function produced before.